mkdocs_awesome_nav/nav/directory.py

```diff
@@ -17,10 +18,16 @@ def __init__(self, directory: Directory, *, parent_config: Optional[NavConfig] =
         self._directory = directory
         self.path = directory.path
         self.config = self._load_config(parent_config)
-        self._title = title or self.config.title or self._generate_title()
+        self._explicit_title = title or self.config.title
+        self._resolved_title: str | None = None
 
     def resolve(self, context: MkdocsFilesContext) -> NavSection | NavPage | list[NavSection | NavPage]:
         context.visit(self._directory)
+
+        # Generate title if not already set
+        if self._resolved_title is None:
```

Comment:
This will always be true. I think you don't actually need two separate fields (`_explicit_title`and `_resolved_title`).

I would revert back to `_title` and then do this:

```py
if self._title is None:
    self._title = self._generate_title(context)
```

Reply:
Refactored